build/src/lib.rs

```diff
@@ -0,0 +1,280 @@
+        let image_base = "ghcr.io/succinctlabs/sp1";
+        format!("{}:{}", image_base, tag)
+    })
+}
+
+/// Modify the command to build the program in the docker container. Mounts the entire workspace
+/// and sets the working directory to the program directory.
+fn get_docker_cmd(
+    command: &mut Command,
+    args: &BuildArgs,
+    workspace_root: &Utf8PathBuf,
+    program_dir: &Utf8PathBuf,
+) -> Result<()> {
+    let image = get_docker_image(&args.tag);
+
+    let docker_check = Command::new("docker")
+        .args(["info"])
+        .stdout(Stdio::null())
+        .stderr(Stdio::null())
+        .status()
+        .context("failed to run docker command")?;
+
+    if !docker_check.success() {
+        eprintln!("docker is not installed or not running: https://docs.docker.com/get-docker/");
+        exit(1);
+    }
+
+    // Mount the entire workspace, and set the working directory to the program dir.
+    let workspace_root_path = format!("{}:/root/program", workspace_root);
+    let program_dir_path = format!(
+        "/root/program/{}",
+        program_dir.strip_prefix(workspace_root).unwrap()
+    );
+    let mut child_args = vec![
+        "run".to_string(),
+        "--rm".to_string(),
+        "--platform".to_string(),
+        "linux/amd64".to_string(),
+        "-v".to_string(),
+        workspace_root_path,
+        "-w".to_string(),
+        program_dir_path,
+        image,
+        "prove".to_string(),
+        "build".to_string(),
+    ];
+
+    // Add the `cargo prove build` arguments to the child process command, while ignoring
+    // `--docker`, as this command will be invoked in a docker container.
+    add_cargo_prove_build_args(&mut child_args, args.clone(), true);
+
+    // Set the arguments and remove RUSTC from the environment to avoid ensure the correct
+    // version is used.
+    command.args(&child_args).env_remove("RUSTC");
+    Ok(())
+}
+
+/// Modify the command to build the program in the local environment.
+fn get_build_cmd(
+    command: &mut Command,
+    args: &BuildArgs,
+    program_dir: &Utf8PathBuf,
+    build_target: String,
+) -> Result<()> {
+    let rust_flags = [
+        "-C".to_string(),
+        "passes=loweratomic".to_string(),
+        "-C".to_string(),
+        "link-arg=-Ttext=0x00200800".to_string(),
+        "-C".to_string(),
+        "panic=abort".to_string(),
+    ];
+
+    let mut cargo_args = vec![
+        "build".to_string(),
+        "--release".to_string(),
+        "--target".to_string(),
+        build_target,
+    ];
+
+    // Add the `cargo prove build` arguments to the child process command, while ignoring
+    // `--docker`, as we already know it's unused in this conditional.
+    add_cargo_prove_build_args(&mut cargo_args, args.clone(), true);
+
+    // Ensure the Cargo.lock doesn't update.
+    cargo_args.push("--locked".to_string());
+
+    command
+        .current_dir(program_dir.clone())
+        .env("RUSTUP_TOOLCHAIN", "succinct")
```

Comment:
I'm confused why this isn't activating...

Reply:
I needed to strip the `RUSTC` environment variable in the command, as it was inheriting the `RUSTC` from the parent build process.